### PR TITLE
Add support for specifying file extensions

### PR DESCRIPTION
**Pull request by ChrisChinchilla**

### CODE REVIEW

**Comment by ChrisChinchilla** [comment]
Care to take a look @emmenko this seems to work based on my tests

File: package.json [hunk 1]
```diff
@@ -60,7 +60,24 @@
                     "type": "string",
                     "default": "vale",
                     "description": "Absolute path to Vale binary. Special var ${workspaceFolder} can be used in path (NOTE: in windows you can use '/' and can omit '.cmd' in path)"
-                }
+                },
+                "vscode-vale.fileExtensions": {
+                    "scope": "resource",
+                    "type": "array",
+                    "default": [
```
Comment (emmenko) [inline package.json:67]: Does it work by specifying the defaults only here and not in the getter?

**Review by emmenko — COMMENTED** [review]
Looks good, just left one question

File: package.json (hunk 1 above)
Comment (ChrisChinchilla) [reply package.json:67]: I don't really understand the question sorry. I pretty much followed the same steps as in the project you linked elsewhere and it seemed to work well and tidily.

Comment (emmenko) [reply package.json:67]: I was referring to the `get('fileExtensions')` part, where you can also pass a default value.

image

But I suppose this works too.

File: src/extension.ts [hunk 2]
```diff
@@ -357,15 +357,9 @@ const runValeOnWorkspace = async (): Promise<ValeDiagnostics> => {
...
-        "md",
-        "markdown",
-        "txt",
-        "rst",
-        "tex",
-        "adoc",
-        "asciidoc",
-    ];
+
+    const extensions: Array<string> = workspace.getConfiguration("vscode-vale").get<string[]>("fileExtensions")!;
```
Comment (emmenko) [inline src/extension.ts]: Nitpick: to let TS infer the return value automatically

```suggestion
    const extensions = workspace
      .getConfiguration("vscode-vale")
      .get<ReadonlyArray<string>>("fileExtensions")!;
```

image

**Review by emmenko — COMMENTED** [review]
Thanks, can you also update the README with the new configuration option?

**Review by emmenko — APPROVED** [review]
💯 Thanks

**Comment by ChrisChinchilla** [comment]
Thanks @emmenko I'll release soon, might overhaul a few read me and changelog things first

**Pull request merged**